process_scans.py

```diff
@@ -0,0 +1,99 @@
+import csv
+import io
+from functools import lru_cache
+
+import boto3
+
+from nessus import download_report, list_scans, prepare_export
+
+
+def find_scans():
+    scans = list_scans()
+    for scan in scans["scans"]:
+        # This is for testing as we haven't run a full scan yet
+        if scan["status"] == "completed":
+            print(f"Preparing export for {scan['name']}")
+            token = prepare_export(scan["id"])
+            csv_text = download_report(token)
+            process_csv(csv_text, scan)
+        elif scan["status"] == "empty":
+            print(f"Scan {scan['name']} has not run.")
+
+
+def process_csv(csv_text, scan):
+    """
+    Send the CSV - 1 row at a time - to cloud watch. This will create a
+    new event for each row in Splunk.
+    """
+    #  Need to use StringIO because csv.reader expects a file object
+    with io.StringIO(csv_text) as f:
+        reader = csv.reader(f)
+
+        group_name = "/gds/nessus-data"
+        stream_name = f"{scan['last_modification_date']}-{scan['name']}"
+        token = create_log_stream(group_name, stream_name)
+
+        events = []
+
+        for row in reader:
+
+            events.append(
+                {
+                    "timestamp": scan["last_modification_date"] * 1000,
+                    "message": ",".join(row).replace("\n", " "),
+                }
+            )
+
+            # Send in batches of 10_000
+            if len(events) >= 9999:
+                token = logs_client().put_log_events(
+                    logGroupName="/gds/nessus-data",
```

Comment:
Don't see why we can't re-use `group_name` here (like we do with `stream_name` on the line below):
```suggestion
                    logGroupName=group_name,
```